metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -1954,6 +1954,42 @@ def emit_datasource(
                     aspect_name=c.UPSTREAM_LINEAGE,
                     aspect=upstream_lineage,
                 )
+        else:
+            # Fetch upstream information from upstreamColumns which are mostly coming
+            # from CUSTOM SQLs when Tableu doesn't provide upstream tables
+            # With this approach, we also join lingering CustomSQL entities to Embedded Data Sources
```

Comment:
copying in your PR description

```suggestion
            # Tableau's metadata graphql API sometimes returns empty an list for upstreamTables
            # for embedded datasources. However, the upstreamColumns field often includes information.
            # This attempts to populate upstream table information from the upstreamColumns field.
```